### PR TITLE
New function/test/filter has_value for templating

### PR DESCRIPTION
**Pull request by ehendrix23**
Add new function/test/filter has_value to check if entity is known and does not have state unknown or unavailable.

## Proposed change

Adding new templating function has_value for checking if an entity is known and not unavailable.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/79550
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/configuration/templating.markdown [hunk 1]
```diff
@@ -123,6 +125,24 @@ Other state examples:
...
+
+{% if sensor.train_departure_time is nominal %}
+  {{ ... }}
+
+
+#check sensor.train_departure_time and sensor.train_arrival_time state
+{% if is_nominal(['sensor.train_departure_time', 'sensor.train_arrival_time']) %}
+  {{ ... }}
+
+{% if ['sensor.train_departure_time', 'sensor.train_arrival_time']|is_nominal() %}
```
Comment (ThomDietrich) [inline source/_docs/configuration/templating.markdown]:
```suggestion
{% if ['sensor.train_departure_time', 'sensor.train_arrival_time'] | is_nominal() %}
```

File: source/_docs/configuration/templating.markdown [hunk 2]
```diff
@@ -76,6 +76,8 @@ Not supported in [limited templates](#limited-templates).
 - `is_state('device_tracker.paulus', 'home')` will test if the given entity is the specified state.
 - `state_attr('device_tracker.paulus', 'battery')` will return the value of the attribute or None if it doesn't exist.
 - `is_state_attr('device_tracker.paulus', 'battery', 40)` will test if the given entity attribute is the specified state (in this case, a numeric value). Note that the attribute can be `None` and you want to check if it is `None`, you need to use `state_attr('sensor.my_sensor', 'attr') == None`. 
+- `is_nominal('sensor.my_sensor')` will test if the given entity exist and is not unknown or unavailable. A list of entities can be provided instead of a string. Can be used as a filter. `nominal` is the test equivalent.
```
Comment (ThomDietrich) [inline source/_docs/configuration/templating.markdown]: After the short discussion in the core PR: Would it make sense to explicitly mention that for a list of entities an OR logic is applied!? Cheers

**Comment by ThomDietrich** [comment]
Lgtm

**Review by ThomDietrich — DISMISSED** [review]
Lgtm

File: source/_docs/configuration/templating.markdown [hunk 3]
```diff
@@ -175,6 +176,13 @@ Other state examples:
 {% if states('sensor.train_departure_time') in ("unavailable", "unknown") %}
   {{ ... }}
 
+{% if has_value('sensor.train_departure_time') %}
+  {{ ... }}
+
+{% if sensor.train_departure_time is has_value %}
```
Comment (frenck) [inline source/_docs/configuration/templating.markdown]: what is `sensor.trains_departure_time`? as in, where does this variable come from?

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_docs/configuration/templating.markdown (hunk 3 above)
Comment (ehendrix23) [reply source/_docs/configuration/templating.markdown]: Sorry, missed states. there. Fixed.

File: source/_docs/configuration/templating.markdown [hunk 4]
```diff
@@ -175,6 +176,13 @@ Other state examples:
 {% if states('sensor.train_departure_time') in ("unavailable", "unknown") %}
   {{ ... }}
 
+{% if has_value('sensor.train_departure_time') %}
+  {{ ... }}
+
+{% if states.sensor.train_departure_time is has_value %}
```
Comment (frenck) [inline source/_docs/configuration/templating.markdown]: `states` should not be used in this context, as per our own documentation:

image

**Review by frenck — CHANGES_REQUESTED** [review]

**Review by frenck — APPROVED** [review]
Thanks, @ehendrix23 👍 

../Frenck

**Pull request merged**